controllers/loadtest_controller.go

```diff
@@ -175,131 +176,221 @@ func (r *LoadTestReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+					poolCapacities[pool] = 0
+				}
 			}
 
-			return ctrl.Result{}, err
+			poolCapacities[pool]++
 		}
 
-		result, err := createPod(pod)
-		if result != nil && !kerrors.IsAlreadyExists(err) {
-			logWithServer.Error(err, "failed to create pod for server")
-
-			test.Status.State = grpcv1.Errored
-			test.Status.Reason = grpcv1.KubernetesError
-			test.Status.Message = fmt.Sprintf("failed to create pod for server at index %d: %v", i, err)
-
-			if updateErr := r.Status().Update(ctx, test); updateErr != nil {
-				logWithServer.Error(updateErr, "failed to update status after failure to create pod for server")
+		poolAvailabilities := make(map[string]int)
+		for pool, capacity := range poolCapacities {
+			poolAvailabilities[pool] = capacity
+		}
+		for _, pod := range pods.Items {
+			pool, ok := pod.Labels[config.PoolLabel]
+			if !ok {
+				log.Info("encountered a pod without a pool label", "pod", pod)
+				continue
+			}
+			if pod.Status.Phase != corev1.PodSucceeded && pod.Status.Phase != corev1.PodFailed {
+				poolAvailabilities[pool]--
 			}
-
-			return *result, err
 		}
-	}
-	for i := range missingPods.Clients {
-		logWithClient := log.WithValues("client", missingPods.Clients[i])
-
-		pod, err := builder.PodForClient(&missingPods.Clients[i])
-		if err != nil {
-			logWithClient.Error(err, "failed to construct a pod struct for supplied client struct")
 
-			test.Status.State = grpcv1.Errored
-			test.Status.Reason = grpcv1.ConfigurationError
-			test.Status.Message = fmt.Sprintf("failed to construct a pod for client at index %d: %v", i, err)
-
-			if updateErr := r.Status().Update(ctx, test); updateErr != nil {
-				logWithClient.Error(updateErr, "failed to update status after failure to construct a pod for client")
+		adjustAvailabilityForDefaults := func(defaultPoolKey, defaultPoolName string) {
+			if c, ok := missingPods.NodeCountByPool[defaultPoolKey]; ok {
+				missingPods.NodeCountByPool[defaultPoolName] += c
 			}
-
-			return ctrl.Result{}, err
+			delete(missingPods.NodeCountByPool, defaultPoolKey)
 		}
+		adjustAvailabilityForDefaults(status.DefaultClientPool, defaultClientPool)
+		adjustAvailabilityForDefaults(status.DefaultDriverPool, defaultDriverPool)
+		adjustAvailabilityForDefaults(status.DefaultServerPool, defaultServerPool)
+
+		for pool, requiredNodeCount := range missingPods.NodeCountByPool {
+			availableNodeCount, ok := poolAvailabilities[pool]
+			if !ok {
+				log.Error(errNonexistentPool, "requested pool does not exist and cannot be considered when scheduling", "requestedPool", pool)
+				test.Status.State = grpcv1.Errored
+				test.Status.Reason = grpcv1.PoolError
+				test.Status.Message = fmt.Sprintf("requested pool %q does not exist", pool)
+				if updateErr := r.Status().Update(ctx, test); updateErr != nil {
+					log.Error(updateErr, "failed to update status after failure due to requesting nodes from a nonexistent pool")
+				}
+				return ctrl.Result{}, errNonexistentPool
```

Comment:
If your intension is not to requeue due to this error you might not want to return this error since whenever an error is return it automatically requeued.

Reply:
Fair point. I still do not understand why `return ctrl.Result{Requeue: false}, errNonexistentPool` does not appear to be an option.

Reply:
well if you do want that, you could log the error, just not to return it.